### DISCUSSION TITLE
During authorization, e.g. Require claims_expr 'jq_filter', is there a way to reference Apache environment variables inside JQ filter.

### DISCUSSION THREAD

**Discussion by Mengchang-Chen** [OpenIDC/mod_auth_openidc#1004] [Q&A]
We are using Require claims_expr 'jq filter' to do the authorization.  The question is if the Apache environment variables can be referenced within the JQ filters, as they do in the Require claim, e.g. Require claim "roles:%{env:MATCH_BACKEND}".  If the answer is yes, can you give some examples?

Thanks.

**Comment by zandbelt** [comment]
[edited]
after looking into the code I do think it *may* be possible, but I don't have a concrete example; I'm not sure how escaping between Apache expressions and JQ filters would have to work

**Reply by Mengchang-Chen** [reply]
Thanks for looking.  I think there may be 2 ways to implement this:

1. Replacing variable placeholder in jq filter with its value before submit the filter to jq compiler;

2. Use setenv() to export named Apache variables in filters to httpd process space and let jq retrieve their values from process environment by using jq builtin function, env():
$ echo '{}' | jq 'env.FOO == "bar"'
false
$ echo '{}' | FOO=bar jq 'env.FOO == "bar"'
true
